content/en/docs/pipelines/sdk/python-function-components.ipynb

```diff
@@ -176,25 +182,141 @@
+    "dependencies installed.\n",
+    "\n",
+    "If a prebuilt container is not available, you can build a custom container\n",
+    "image with your Python function's dependencies. For more information about\n",
+    "building a custom container, read the [Dockerfile reference guide in the Docker\n",
+    "documentation][dockerfile].\n",
+    "\n",
+    "If you build or select a container image, instead of using the default\n",
+    "container image, the container image must use Python 3.5 or later.\n",
+    "\n",
+    "<a name=\"pass-data\"></a>\n",
+    "### Understanding how data is passed between components\n",
+    "\n",
+    "When Kubeflow Pipelines runs your component, a container image is started in a\n",
+    "Kubernetes Pod and your component's inputs are passed in as command-line\n",
+    "arguments. When your component has finished, the component’s outputs are\n",
+    "returned as files.\n",
+    "\n",
+    "Python function-based components make it easier to build pipeline components by\n",
+    "building the component specification for you. Python function-based components\n",
+    "also handle the complexity of passing inputs into your component and passing\n",
+    "your function's outputs back to your pipeline.  \n",
+    "\n",
+    "The following sections describe how to pass parameters by value and by file. \n",
+    "\n",
+    "*   Parameters that are passed by value include numbers, booleans, and short\n",
+    "    strings. Kubeflow Pipelines passes parameters to your component by value,\n",
+    "    by passing the values as command-line arguments.\n",
+    "*   Parameters that are passed by file include CSV, images, and complex types.\n",
+    "    These files are stored in a location that is accessible to your component\n",
+    "    running on Kubernetes, such as a persistent volume claim or a cloud\n",
+    "    storage service. Kubeflow Pipelines passes parameters to your component by\n",
+    "    file, by passing their paths as a command-line argument.\n",
+    "\n",
+    "<a name=\"pass-by-value\"></a>\n",
+    "#### Passing parameters by value\n",
+    "\n",
+    "Python function-based components make it easier to pass parameters between\n",
+    "components by value (such as numbers, booleans, and short strings), by letting\n",
+    "you define your component’s interface by annotating your Python function. The\n",
+    "supported types are `int`, `float`, `bool`, and `string`. If you do not\n",
+    "annotate your function, these input parameters are passed as strings.\n",
+    "\n",
+    "If your component returns multiple outputs by value, annotate your function\n",
+    "with the [`typing.NamedTuple`][named-tuple-hint] type hint and use the\n",
+    "[`collections.namedtuple`][named-tuple] function to return your function's\n",
+    "outputs as a new subclass of `tuple`.\n",
+    "\n",
+    "You can also return metadata and metrics from your function.\n",
```

Comment:
nit: Can we use UI metadata here and below. Metadata itself can be ambiguous for it also refers to the artifact metadata, for example.

Reply:
The term "UI metadata" doesn't appear to be in use elsewhere in the docs. Would it be alright to discuss this and address this issue in another PR?

Reply:
Sounds good. Thanks